src/http/controllers/products-controllers.ts

```diff
@@ -1,26 +1,39 @@
 import {Product} from "../../models/product";
-import * as queries from "../../db/queries";
+import * as queries from "../../db/db-queries";
+import * as UserCart from '../../socket/cache';
 
 export const getProducts = async (ctx, next) => {
-    ctx.ok(await queries.findProductsQuery());
+    ctx.products = await queries.findProductsQuery();
     await next();
 };
 
+export const getUpdatedProducts = async (ctx, next) => {
+    let products = ctx.products;
+    const usersProducts = UserCart.getUsersProducts();
+    products = products.map(dbProduct => {
+        let productAmount: number = 0;
+        Object.keys(usersProducts).filter(userId => usersProducts[userId][dbProduct._id.toString()]).forEach(userId =>
```

Comment:
move to method inside cache file ... method that calculates product's left amount